### PR TITLE
Generatestops fix

### PR DESCRIPTION
**Pull request by kahummer**
- Fix timeseries stops generation bug
- Comma separated list style
Fix: https://github.com/onaio/irfsomalia/issues/890
- [x] Add legend docs

- [x] Add legend test

### CODE REVIEW

File: src/lib/components/Filter/Filter.js [hunk 1]
```diff
@@ -330,6 +330,10 @@ export class Filter extends Component {
     // determine whether to update the compnent state
     const doUpdate =
       (layerId !== this.state.layerId && filterOptions && Object.keys(filterOptions).length > 0) ||
+      // (nextProps.layerObj &&
```
Comment (kelvin-muchiri) [inline src/lib/components/Filter/Filter.js]: Let's remove commented code?

File: docs/features/menu-legend-links.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+### Site level menu (sector/subsector)
+##### Sector level spec
+The `link` and `description` fields are optional one can be provided another ommitted as desired
+```
+"hyperLink": {
+        "Boundaries": {
+          "description": "desc",
+          "link": "https://gisida.com",
+        },
+        "BRCiS II Output Indicators": {
```
Comment (kelvin-muchiri) [inline docs/features/menu-legend-links.md:19]: I remember some keys here consisted of the parent menu name and the sub-menu menu name? Should we mention thus in the documentation?

File: docs/features/menu-legend-links.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Menu/legend info Links
+gisida-react menu component builds menu items/sections based on json spec provided on clients site-config file. Menu/legend info links feature seeks to give context of the selected layer, menu sections and subsections. The links can be added by adding links and descriptions to layer/siteconfig as shown below.    
+
```
Comment (kelvin-muchiri) [inline docs/features/menu-legend-links.md:5]: Could  we add an image/preview of the feature here?

File: src/lib/components/HyperLink/HyperLink.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const HyperLink = (props) => {
```
Comment (kelvin-muchiri) [inline src/lib/components/HyperLink/HyperLink.js:11]: Do we need to add how to use this comment in the documentation?

File: src/lib/components/Layer/tests/Layer.test.js [hunk 5]
```diff
@@ -14,20 +14,31 @@ const mapId = 'map-1';
...
+        visible: true,
+        category: 'Boundaries and Place Labels',
+        id: 'sample-layer',
+        zoomOnToggle: true,
+      },
+    },
   },
   APP,
   LOC: {},
+  // MAP: {layers: [{}, {}]}
```
Comment (kelvin-muchiri) [inline src/lib/components/Layer/tests/Layer.test.js]: Let's removed this commented code

File: docs/features/menu-legend-links.md (hunk 2 above)
Comment (kahummer) [reply docs/features/menu-legend-links.md:19]: I think the parent layer should be added to the sub-menu and sub-categories only. Here is the section that explains that
https://github.com/onaio/gisida-react/pull/216/files#diff-e36307f8ee41ec0bb1f69c74262417f8639c5b6cf890dc7af5dbd74cc1592773R25-R33

File: src/lib/components/HyperLink/HyperLink.js (hunk 4 above)
Comment (kelvin-muchiri) [reply src/lib/components/HyperLink/HyperLink.js:11]: @kahummer Sorry, I meant **component**. DO we have documentation for this component (what it does, how to import and use) Looks like it should be part of Menu/legend info Links documentation?

**Review by kelvin-muchiri — APPROVED** [review]

**Pull request merged**